### PR TITLE
Ignore unused_element_parameter

### PR DESCRIPTION
**Pull request by srawlins**
When the analyzer detects an unused parameter, it reports `unused_element` which can be ignored with an inline ignore like `// ignore: unused_element`. The analyzer will start reporting instead, `unused_element_parameter`. There are ignores in flutter/flutter that need to be updated to the new code. In order to incrementally migrate, they can be changed to `// ignore: unused_element, unused_element_parameter`.

After flutter/flutter is using a new enough analyzer, we can then change those to `// ignore: unused_element_parameter`.

Work towards https://github.com/flutter/flutter/issues/126924

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [ ] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [ ] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/material/about.dart [hunk 1]
```diff
@@ -1052,9 +1052,9 @@ class _MasterDetailFlow extends StatefulWidget {
   const _MasterDetailFlow({
     required this.detailPageBuilder,
     required this.masterViewBuilder,
-    this.automaticallyImplyLeading = true, // ignore: unused_element
+    this.automaticallyImplyLeading = true, // ignore: unused_element, unused_element_parameter
     this.detailPageFABlessGutterWidth,
-    this.displayMode = _LayoutMode.auto, // ignore: unused_element
+    this.displayMode = _LayoutMode.auto, // ignore: unused_element, unused_element_parameter
```
Comment (goderbauer) [inline packages/flutter/lib/src/material/about.dart:1057]: Unrelated to your PR: We should probably just remove these parameters on _MasterDetailFlow.

**Review by goderbauer — APPROVED** [review]
LGTM

File: packages/flutter/lib/src/material/about.dart (hunk 1 above)
Comment (srawlins) [reply packages/flutter/lib/src/material/about.dart:1057]: I'd be OK with that too 😁

Comment (goderbauer) [reply packages/flutter/lib/src/material/about.dart:1057]: https://github.com/flutter/flutter/pull/126935

**Comment by goderbauer** [comment]
https://github.com/flutter/flutter/pull/126935 is merged. Can you rebase this on top of that?

**Pull request merged**